test/parallel/test-worker-init-failure.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const child_process = require('child_process');
+
+// Test that workers fail with meaningful error message
+// when their initialization fails.
+
+if (common.isWindows) {
+  common.skip('ulimit does not work on Windows.');
+}
+
+// A reasonably low fd count. An empty node process
+// creates around 30 fds for its internal purposes,
+// so making it too low will crash the process early,
+// making it too high will cause too much resource use.
+const OPENFILES = 128;
+
+// Double the open files - so that some workers fail for sure.
+const WORKERCOUNT = 256;
+
+if (process.argv[2] === 'child') {
+  const { Worker } = require('worker_threads');
+  for (let i = 0; i < WORKERCOUNT; ++i) {
+    const worker = new Worker(
+      'require(\'worker_threads\').parentPort.postMessage(2 + 2)',
+      { eval: true });
+    worker.on('message', (result) => {
+      assert.strictEqual(result, 4);
+    });
+    worker.on('error', (e) => {
+      assert.match(e.message, /Worker initialization failure: EMFILE/);
+      assert.strictEqual(e.code, 'ERR_WORKER_INIT_FAILED');
+    });
+  }
```

Comment:
IIUC this test would still pass if all workers succeed. I think `common.mustCallAtLeast` around `worker.on('error',`'s callback should solve this.

Reply:
@lundibundi - as explained in https://github.com/nodejs/node/pull/31929#discussion_r384320438 , we cannot guarantee how many workers will error, and is highly system-dependant. So the only thing we can do is to make sure the error is of expected type, if at all they fail. Are you fine with that?

Reply:
`mustCallAtLeast` will guarantee that at least (i.e. 1 by default) calls are made there is no need for a precise number and that's exactly what we need.

Reply:
consider this scenario: An AIX system with 1 CPU. In AIX, the child thread gets the rest of the scheduling CPU slot, soon after the creation.
When we run this test there, it is possible that all threads run successfully, one at a time. (we already seen a failure in AIX)

Reply:
Well, we can always check the cpu count on the system and actually just `skip` the test in that case. There is no point in running the test if it doesn't test anything.

Reply:
That is an extreme scenario I used for demonstrating the scenario. In real cases, even if there are CPUs, based on the load in the system, similar situations are possible.
So in short, the intent of the test case is:
 - run with a constraint resource limit on fds
 - spawn few workers whose count is above that limit
 - select the fd count:worker count ratio based on a reasonably loaded system
 - if any of the workers fail, make sure there is an error callback to catch it
 - make sure the error is one of the known type.

Reply:
Ok, I got it, so the idea is to make sure that if there is an error while running in a constrained environment it will be an expected error and not some other one or i.e. crash.

Could you add a note (probably near 'error' event handler) about this so that people won't be confused later on and try to add some variant of `mustCall*` there?

Reply:
@lundibundi, there are 3 variants for `mustCall*` namely, `mustCall`, `mustNotCall` and `mustCallAtleast`. None of them are suited in this case.

Reply:
Yeah, but that is not clear from the code-only perspective, that's why I suggested adding a comment in the lines of
``` // We want to test that if there is an error in a constrained running environment it will be one of [...] . `common.mustCall*` cannot be used here as in some environments (i.e. single cpu) `ulimit` won't lead to such an error. ```
somewhere near the `.on('error'` handler.

Reply:
@lundibundi, thanks. Added the suggested comment.